debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/legacy/MySqlConnectorTask.java

```diff
@@ -192,9 +191,8 @@ else if (!earliestBinlogFilename.endsWith("00001")) {
             if (startWithSnapshot) {
                 // We're supposed to start with a snapshot, so set that up ...
                 SnapshotReader snapshotReader = new SnapshotReader("snapshot", taskContext);
-                if (snapshotEventsAsInserts) {
-                    snapshotReader.generateInsertEvents();
```

Comment:
I think this method should also be removed. There should be no means of emitting snapshot records as insert events within the connector. The SMT should be the single point in the code base allowing for that.

Reply:
Agreed. But there are many tests relying on `generateInsertEvents()` to emit snapshots as `CREATE`.
Should they be now emitted as `READ`?

Reply:
Yes, exactly. Generally speaking, any test relying on the old behavior should configure the SMT.